Library/Koala/KoalaTrackingClient.swift

```diff
@@ -99,59 +99,59 @@ public final class KoalaTrackingClient: TrackingClientType {
     }
   }
 
-  private func load() {
-    dispatch_async(self.queue) {
+  fileprivate func load() {
+    self.queue.async {
       guard
-        let file = self.fileName()
-        where NSFileManager.defaultManager().fileExistsAtPath(file),
-        let buffer = NSKeyedUnarchiver.unarchiveObjectWithFile(file) as? [[String: AnyObject]]
+        let file = self.fileName(), FileManager.default.fileExists(atPath: file),
+        let buffer = NSKeyedUnarchiver.unarchiveObject(withFile: file) as? [[String:Any]]
         else { return }
 
       self.buffer = buffer + self.buffer
 
-      _ = try? NSFileManager.defaultManager().removeItemAtPath(file)
+      _ = try? FileManager.default.removeItem(atPath: file)
     }
   }
 
-  private static func base64Payload(payload: [AnyObject]) -> String? {
-    return (try? NSJSONSerialization.dataWithJSONObject(payload, options: []))
-      .map { $0.base64EncodedStringWithOptions([]) }
+  fileprivate static func base64Payload(_ payload: [Any]) -> String? {
+    return (try? JSONSerialization.data(withJSONObject: payload, options: []))
+      .map { $0.base64EncodedString(options: []) }
   }
 
-  private func koalaURL(dataString: String) -> NSURL? {
+  fileprivate func koalaURL(_ dataString: String) -> URL? {
     #if DEBUG
       if dataString.characters.count >= 10_000 {
         print("[Koala Error] Base64 payload is longer than 10,000 characters.")
       }
     #endif
-    return NSURL(string: "\(self.endpoint.base)?data=\(dataString)")
+    return URL(string: "\(self.endpoint.base)?data=\(dataString)")
   }
 
-  private static func koalaRequest(url: NSURL) -> NSURLRequest {
-    let request = NSMutableURLRequest(URL: url)
-    request.HTTPMethod = "POST"
+  fileprivate static func koalaRequest(_ url: URL) -> URLRequest {
+    var request = URLRequest(url: url)
+    request.httpMethod = "POST"
```

Comment:
nice to be able to just use `var request = ` instead of a whole other type `NSMutableURLRequest`.